catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FunctionFilterNode.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.ui.forms.model;
+
+import static org.apache.commons.lang3.Validate.notNull;
+
+import java.util.Map;
+import java.util.Optional;
+import javax.annotation.Nullable;
+import org.boon.json.annotations.JsonProperty;
+import org.codice.ddf.catalog.ui.forms.api.FilterNode;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class FunctionFilterNode implements FilterNode {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(FunctionFilterNode.class);
+
+  private static final ThreadLocal<ValueVisitor> VALUE_VISITOR_THREAD_LOCAL =
```

Comment:
Same comment here regarding the use of `ThreadLocal`.

Reply:
fixed